agent/checks/grpc.go

```diff
@@ -69,8 +67,8 @@ func (probe *GrpcHealthProbe) Check(target string) error {
 	if err != nil {
 		return err
 	}
-	if response == nil || response.Status != hv1.HealthCheckResponse_SERVING {
-		return ErrGRPCUnhealthy
+	if response.Status != hv1.HealthCheckResponse_SERVING {
+		return fmt.Errorf("gRPC %s serving status: %s", target, response.Status)
```

Comment:
response.Status is an int, might be improved:
```go
statusTxt, ok := hv1.HealthCheckResponse_ServingStatus_name[response.Status]
if !ok {
   statusTxt = fmt.Sprintf("UnexpectedGRPCStatus:=%d", response.Status)
}
return fmt.Errorf("gRPC Error: %s for target %s", statusTxt, target)
```

Reply:
`response.Status`  already has `String` method, that's why I put `%s` there.

```
func (m *HealthCheckResponse) String() string { return proto.CompactTextString(m) }
```

Reply:
ok, sorry :-)

Reply:
I think `UnexpectedGRPCStatus` can confuse people with general gRPC status codes.